features/enlinkd/generator/src/test/java/org/opennms/enlinkd/generator/TopologyGeneratorTest.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.enlinkd.generator;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+
+import java.util.Set;
+
+import org.junit.Test;
+import org.opennms.netmgt.dao.api.GenericPersistenceAccessor;
+import org.opennms.netmgt.topologies.service.api.OnmsTopology;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyConsumer;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyDao;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyEdge;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyMessage;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyPort;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyRef;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyUpdater;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyVertex;
+
+public class TopologyGeneratorTest {
+    private int numVerticesUpdated = 0;
+    private int numPortsUpdated = 0;
+    private int numEdgesUpdated = 0;
+
+    /**
+     * This test verifies that any topology generated by the topology generator is sent to the topology DAO as a series
+     * of topology updates.
+     */
+    @Test
+    public void shouldUpdateTopologyDao() {
+        int numNodes = 10;
+        int numPorts = 180;
+        int numLinks = 10;
+        TopologyGenerator.ProgressCallback mockProgressCallback = mock(TopologyGenerator.ProgressCallback.class);
+        TopologyGenerator generator = TopologyGenerator.builder()
+                .persister(new TopologyPersister(mock(GenericPersistenceAccessor.class), mockProgressCallback))
+                .progressCallback(mockProgressCallback)
+                .build();
+        TopologySettings settings = TopologySettings.builder()
+                .amountElements(numNodes)
+                .amountIpInterfaces(20)
+                .amountLinks(numLinks)
+                .amountNodes(numNodes)
+                .amountSnmpInterfaces(numPorts)
+                .protocol(TopologyGenerator.Protocol.ospf)
+                .topology(TopologyGenerator.Topology.ring)
+                .topologyDao(new TestOnmsTopologyDao())
+                .build();
+
+        generator.generateTopology(settings);
+
+        assertThat(numVerticesUpdated, is(equalTo(numNodes)));
+        assertThat(numPortsUpdated, is(equalTo(numPorts)));
+        assertThat(numEdgesUpdated, is(equalTo(numLinks / 2)));
+    }
+
+    public class TestOnmsTopologyDao implements OnmsTopologyDao {
```

Comment:
If most of the code is not used, maybe consider using a `EasyMock.createNiceMock(...)` as it does the same and is less verbose.

Reply:
This test class has been removed.